src/mon/MgrMonitor.cc

```diff
@@ -892,10 +902,16 @@ bool MgrMonitor::promote_standby()
   }
 }
 
-void MgrMonitor::drop_active()
+bool MgrMonitor::drop_active()
 {
   ceph_assert(mon.osdmon()->is_writeable());
 
+  bool plugged = false;
+  if (!paxos.is_plugged()) {
+    paxos.plug();
```

Comment:
Wouldn't it be enough to plug() before we propose OSD map with the the blocklisted entries (line 931)  and unplug right after?
```
paxos.plug()
request_proposal(mon.osdmon());
paxos.unplug();
```

I also noticed that typical usage of plugging and unpluggind  paxos proposals is followed by setting `force_immediate_propose()` . Do we want to set  `need_immediate_propose` flag?

Reply:
The intent here is to prevent a PAXOS transaction before the MgrMap changes are complete and a propose is indicated. The goal being to batch both map changes in one transaction.

Reply:
I don't know about force_immediate_propose. It's not clear to me if it's really necessary.

Reply:
> The intent here is to prevent a PAXOS transaction before the MgrMap changes are complete and a propose is indicated. The goal being to batch both map changes in one transaction.

I understand that we want to batch the OSDMap update and MgrMap updates in one PAXOS transaction.

My question is why is it not sufficient just to plug the OSDMonitor's proposal of updating the OSDMap with the new blocklist, i.e., `request_proposal(mon.osdmon());` in MgrMap::drop_active(), and then unplug right after. During the next MgrMonitor's proposal, both the MgrMap updates (dropping the failed mgr ID from the MgrMap),  and the OSDMap updates will be batched together in a PAXOS transaction, because currently the `propose_pending()` is called only after updates to the pending Mgrmap. Maybe plugging paxos before calling drop_active() and unplugging after is sufficient?

Reply:
To be honest, I'm not sure. It made sense to me that keeping PAXOS plugged until both `PaxosService`s call `propose_pending`.